src/gsd.js

```diff
@@ -100,8 +101,8 @@ function gsd(x, yIn, options) {
     // By the intermediate value theorem We cannot find 2 consecutive maximum or minimum
     for (let i = 1; i < Y.length - 1; ++i) {
         // Minimum in first derivative
-        if ((dY[i] < dY[i - 1]) && (dY[i] <= dY[i + 1]) ||
-            (dY[i] <= dY[i - 1]) && (dY[i] < dY[i + 1])) {
+        if ((dY[i] - dY[i - 1] <  -options.derivativeThreshold) && (dY[i] - dY[i + 1] <= -options.derivativeThreshold) ||
```

Comment:
Still the threlhold in the difference of the first derivative, which is similar to a threshold in the second derivative. Why you dont simply use: 

``if(Math.abs(dY[i])<options.derivativeThreshold)``